corsikaio/subblocks/dtypes.py

```diff
@@ -0,0 +1,20 @@
+from collections import namedtuple, defaultdict
+import numpy as np
+
+
+Field = namedtuple('Field', ['position', 'name', 'shape', 'dtype'])
+
+# set defaults for shape and dtype, this works for the 2 right most attributes
+Field.__new__.__defaults__ = (1, 'f4')
+
+
+def build_dtype(fields, itemsize=4 * 273):
```

Comment:
Do we really need the developer useing `build_type()` to calculate the itemsize? Can't we deduce it from the fields? What a pity. Looking at the examples of `np.dtype` here: https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.dtype.html

```
Using dictionaries. Two fields named ‘gender’ and ‘age’:

>>> np.dtype({'names':['gender','age'], 'formats':['S1',np.uint8]})
dtype([('gender', '|S1'), ('age', '|u1')])
```

It looks like itemsize is not needed. Without testing any of the following I assume `itemsize` is needed when one wants to use `offsets`. Now looking at this, I believe the way to calculate the offsets will fail as soon as one of the fields is not an `f4` or an `i4` or an `S4` ... So ... hm ... maybe ... this is all fine for here ... just the reusability of this function for other projects is limited...

devs must be a bit careful not to assume too much from this function .. but this is what tests are for .. so fine with me.

Reply:
Sorry for the noise.

Reply:
The problem is, when there is unused space at the end of a block. This is the case for run end, event end and event header blocks.

The reading ignores these unused fields, but then the itemsize is not correct. 
As it is known, that each of these corsika blocks is 273 floats, i made it the default.

Reply:
Let's say we each row is 3 floats, but the last one is unused.
having, only using 
```
>>> dt = np.dtype(dict(names=['x', 'y'], formats=['f4', 'f4']))
>>> dt.itemsize
8
```
This is wrong, as it does not take the third, unused field into account, so we have to explicitly set `itemsize=12`
```